client/landing/stepper/declarative-flow/site-setup-flow.ts

```diff
@@ -57,15 +57,13 @@ export const siteSetupFlow: Flow = {
 	useStepNavigation( currentStep, navigate ) {
 		const intent = useSelect( ( select ) => select( ONBOARD_STORE ).getIntent() );
 		const startingPoint = useSelect( ( select ) => select( ONBOARD_STORE ).getStartingPoint() );
-		const siteSlug = useSiteSlugParam();
-		const siteId = useSelect(
-			( select ) => siteSlug && select( SITE_STORE ).getSiteIdBySlug( siteSlug )
-		);
+		const site = useSite();
+		const siteSlug = site ? new URL( site.URL ).host : null;
 		const adminUrl = useSelect(
-			( select ) => siteSlug && select( SITE_STORE ).getSiteOption( siteId as number, 'admin_url' )
+			( select ) => site && select( SITE_STORE ).getSiteOption( site.ID as number, 'admin_url' )
```

Comment:
Since you checked the `site` before calling select, ID is a number for sure.

```suggestion
			( select ) => site && select( SITE_STORE ).getSiteOption( site.ID, 'admin_url' )
```